components/raftstore/src/store/fsm/peer.rs

```diff
@@ -5805,6 +5919,55 @@ fn new_compact_log_request(
     request
 }
 
+fn demote_failed_voters_request(
+    region: &metapb::Region,
+    peer: &metapb::Peer,
+    failed_voters: Vec<metapb::Peer>,
+) -> Option<RaftCmdRequest> {
+    let failed_voter_ids = HashSet::from_iter(failed_voters.iter().map(|voter| voter.get_id()));
```

Comment:
it doesn't make any sense, you should get ids of region

Reply:
Thanks for catching this